101-machine-learning-create/azuredeploy.json

```diff
@@ -2,257 +2,641 @@
+      "type": "Microsoft.Network/virtualNetworks/subnets",
+      "apiVersion": "2019-09-01",
+      "name": "[concat(variables('vnetName'), '/', variables('subnetName'))]",
+      "dependsOn": [
+        "[variables('vnet')]"
+      ],
+      "properties": {
+        "addressPrefix": "[parameters('subnetPrefix')]",
+        "privateEndpointNetworkPolicies": "Disabled",
+        "privateLinkServiceNetworkPolicies": "Enabled",
+        "serviceEndpoints": [
+          {
+            "service": "Microsoft.Storage"
+          },
+          {
+            "service": "Microsoft.KeyVault"
+          },
+          {
+            "service": "Microsoft.ContainerRegistry"
+          }
         ]
+      }
+    },
+    {
+      "condition": "[equals(parameters('storageAccountOption'), 'new')]",
+      "type": "Microsoft.Storage/storageAccounts",
+      "apiVersion": "2019-04-01",
+      "name": "[variables('storageAccountName')]",
+      "location": "[if(variables('isLocationValid'), parameters('location'), 'InvalidLocation')]",
+      "sku": {
+        "name": "[parameters('storageAccountType')]"
       },
-      "storageAccountResourceGroupName": {
-        "type": "string",
-        "defaultValue": "[resourceGroup().name]"
-      },
-      "keyVaultNewOrExisting": {
-        "type": "string",
-        "defaultValue": "new",
-        "allowedValues": [
-          "new",
-          "existing"
-        ],
-        "metadata": {
-          "description": "Determines whether or not a key vault should be provisioned."
+      "kind": "StorageV2",
+      "properties": {
+        "encryption": {
+          "services": {
+            "blob": {
+              "enabled": true
+              },
+            "file": {
+              "enabled": true
+            }
+          },
+          "keySource": "Microsoft.Storage"
+        },
+        "supportsHttpsTrafficOnly": true
+    }
+    },
+    {
+      "condition": "[equals(parameters('storageAccountBehindVNet'), 'true')]",
+      "type": "Microsoft.Resources/deployments",
+      "apiVersion": "2017-05-10",
+      "name": "UpdateStorageAcls",
+      "dependsOn": [
+          "[variables('subnet')]",
+          "[variables('storageAccount')]"
+      ],
+      "resourceGroup": "[resourceGroup().name]",
+      "properties": {
+        "mode": "Incremental",
+        "template": {
+          "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+          "contentVersion": "1.0.0.0",
+          "resources": [
+            {
+              "type": "Microsoft.Storage/storageAccounts",
+              "apiVersion": "2019-04-01",
+              "name": "[variables('storageAccountName')]",
+              "location": "[if(variables('isLocationValid'), parameters('location'), 'InvalidLocation')]",
+              "properties": "[union(reference(variables('storageAccount'), '2019-04-01'), variables('defaultNetworkAcls'))]"
```

Comment:
if this is updating the existing storage account it will replace current ACLs, which would be bad...  if necessary for this sample to work behind a vnet, you'll need to remove the option for an existing storage account

Reply:
It's the requirement for updating existing storage account.

Reply:
It's fine to use an existing storageAccount but you can't modify it in this way, it will remove existing configuration which will break other resources relying on it

Reply:
I made one more change on this kind of update.